app/src/main/java/com/rentminder/service/MembersService.kt

```diff
@@ -9,11 +9,18 @@ import kotlinx.coroutines.withContext
 import retrofit2.awaitResponse
 
 class MembersService {
-    suspend fun fetchMembersList() : List<Members> {
-        return withContext(Dispatchers.IO){
-            val service = RetrofitClientInstance.retrofitInstance?.create(MembersDAO::class.java)
-            val members = async {service?.getAllMembers()?.awaitResponse()?.body() }
-            return@withContext members.await()?: emptyList()
+    private val service: MembersDAO by lazy {
+        RetrofitClientInstance.retrofitInstance?.create(MembersDAO::class.java)
+            ?: throw IllegalStateException("RetrofitClientInstance is null")
+    }
```

Comment:
Implemented Lazy here to initialize the RetrofitClientInstance. This is so its only initialized one time, instead of every time the 'fetchMembersList()' is called.